Don't skip `Option` for nullable collections

fengalin opened the pull request:
Consequence of https://github.com/gtk-rs/gir/pull/1622

fengalin committed:
gir: update for PR#1622

https://github.com/gtk-rs/gir/pull/1622

sdroege commented on gio/src/auto/functions.rs:
```diff
@@ -181,9 +181,9 @@ pub fn content_type_get_symbolic_icon(type_: &str) -> Icon {
 pub fn content_type_guess(
     filename: Option<impl AsRef<std::path::Path>>,
-    data: &[u8],
+    data: Option<&[u8]>,
```
Needs rebasing as this is now manually implemented. Can be autogenerated again :)

sdroege replied:
It was only reverted in the 0.20 branch. main still has it and should have it with the `Option`.

bilelmoussaoui commented on gdk-pixbuf/src/auto/pixbuf_format.rs:
```diff
@@ -29,9 +29,9 @@ impl PixbufFormat {
+    pub fn extensions(&self) -> Option<Vec<glib::GString>> {
         unsafe {
-            FromGlibPtrContainer::from_glib_full(ffi::gdk_pixbuf_format_get_extensions(
+            TryFromGlibPtrContainer::try_from_glib_full(ffi::gdk_pixbuf_format_get_extensions(
```
I would instead name it, MaybeFromGlibPtrContainer or so. Try as a prefix assumes we would get a Result back instead of an Option

bilelmoussaoui replied:
I am also not sure if all those returned `Option<Vec<T>>` make sense or some of them needs annotations fixes to return just `Vec<T>` that would be empty if there are no items.

fengalin replied:
> I would instead name it, MaybeFromGlibPtrContainer or so. Try as a prefix assumes we would get a Result back instead of an Option

Indeed. done.

> I am also not sure if all those returned Option<Vec<T>> make sense or some of them needs annotations fixes to return just Vec<T> that would be empty if there are no items.

Yes, I went looking at the documentation for these functions and spotted some discrepancies last time I tackled this. I'll force annotations with my conclusions in a bit. I just wanted to rebase & push now that I have something usable as a whole (last time I paused, I had just figured out I had to implement all those `Maybe` traits).

fengalin committed:
glib: add MaybeFromGlib traits for Container

C containers can be NULL, which is represented by None in Rust.

This commit:

* adds container specific `MaybeFromGlib` traits which return an Option,
* implements them for the existing containers,
* base the container specific `FromGlib` implementations off the `MaybeFromGlib`
  implementations.

fengalin committed:
Manual updates for gir PR#1622

* [`gdk_pixbuf_format_get_extensions`]
  > The array is `NULL`-terminated.
* [`gdk_pixbuf_format_get_mime_types`]
  > The array is `NULL`-terminated.
* [`g_dbus_proxy_get_cached_property_names`]
  > [...] `NULL` if the value is not in the cache.
* [`g_file_info_list_attributes`]
  > [...] `NULL` on error.
  ==> but NULL as an error is only used for the case where `FileInfo` is NULL, which is not possible in Rust.
* [`g_tls_certificate_get_dns_names`]
  > [...] `NULL` if it’s not available.
* [`g_tls_certificate_get_ip_addresses`]
  > [...] `NULL` if it’s not available.
* [`pango_font_get_languages`]
  > The array is `NULL`-terminated.
* [`pango_font_face_list_sizes`]
  > This is only applicable to bitmap fonts. For scalable fonts, stores `NULL` at the location pointed to by sizes and `0` at the location pointed to by `n_sizes`.
* [`pango_language_get_preferred`](https://docs.gtk.org/Pango/type_func.Language.get_preferred.html)
  > The array is `NULL`-terminated.

[`gdk_pixbuf_format_get_extensions`]: https://docs.gtk.org/gdk-pixbuf/method.PixbufFormat.get_extensions.html
[`gdk_pixbuf_format_get_mime_types`]: https://docs.gtk.org/gdk-pixbuf/method.PixbufFormat.get_mime_types.html
[`g_dbus_proxy_get_cached_property_names`]: https://docs.gtk.org/gio/method.DBusProxy.get_cached_property.html
[`g_file_info_list_attributes`]: https://docs.gtk.org/gio/method.FileInfo.list_attributes.html
[`g_tls_certificate_get_dns_names`]: https://docs.gtk.org/gio/method.TlsCertificate.get_dns_names.html
[`g_tls_certificate_get_ip_addresses`]: https://docs.gtk.org/gio/method.TlsCertificate.get_ip_addresses.html
[`pango_font_get_languages`]: https://docs.gtk.org/Pango/method.Font.get_languages.html
[`pango_font_face_list_sizes`]: https://docs.gtk.org/Pango/method.FontFace.list_sizes.html
[`pango_language_get_preferred`]: https://docs.gtk.org/Pango/type_func.Language.get_preferred.html

fengalin committed:
regen: update for gir PR#1622

https://github.com/gtk-rs/gir/pull/1622

fengalin commented:
Postponing these changes to a future Rust bindings generator rewrite
